File: src/mantle/compat/minecraft/crash/CallableUnsuppConfig.scala
package mantle.compat.minecraft.crash

import cpw.mods.fml.common.ICrashCallable

/**
 * ICrashCallable for mods to report an unsupported environment during crashes
 *
 * @param modIds List of guilty mod IDs
 * @param modLabel The name of the mod creating the callable
 */
class CallableUnsuppConfig(modIds:List[String], modLabel:String) extends ICrashCallable {

  def getLabel = modLabel

  def call = {
    var str: String = "DO NOT REPORT THIS CRASH! Unsupported mods in environment: "
    var firstEntry: Boolean = true
    for (id <- modIds) {
      str = str + (if (firstEntry) id else ", " + id)
      firstEntry = false
    }
    str
  }

}

File: src/mantle/crash/CallableUnsuppConfig.java
package mantle.crash;

import cpw.mods.fml.common.ICrashCallable;

import java.util.List;

// Crash handler for when an unsupported mod is running alongside us
public class CallableUnsuppConfig implements ICrashCallable
{

    private String modId;
    private List<String> modIds;

    public CallableUnsuppConfig(String modId, List<String> modIds)
    {
        this.modId = modId;
        this.modIds = modIds;
    }

    @Override
    public String getLabel ()
    {
        return modId + " Environment";
    }

    @Override
    public String call () throws Exception
    {
        String str = "DO NOT REPORT THIS CRASH! Unsupported mods in environment: ";
        Boolean firstEntry = true;
        for (String id : modIds)
        {
            str = str + (firstEntry ? id : ", " + id);
            firstEntry = false;
        }

        return str;
    }

}